### PR TITLE
Allow a user to override the ssh key data.

### PR DESCRIPTION
**Pull request by squaresurf**
This allows a user to set the ssh key data that they would like to use
as well as tell the fog_driver to not read the key data. For example the
following tells the driver to not load the key data and allows your ssh
key negotiation to be done with an ssh agent where you may have already
unlocked your keys:

```ruby
with_machine_options(ssh_options: { key_data: [] })
```

### CODE REVIEW

**Comment by jkeiser** [comment]
Cool, I'll get this merged shortly.  @squaresurf can you tell me some more details about the use case?  Is this information that is known when we are parsing config.rb in most cases, or is this something where the keys might get unlocked during the actual recipes before you create the machine?

**Comment by jkeiser** [comment]
Merged, 9d5e9db

**Comment by squaresurf** [comment]
tldr; Tell chef-provisioning not to worry about ssh key and just try to connect to the server.

In my use case scenario, I have keys named differently than the key names on the server and I have them loaded in an ssh-agent. So I just need chef-provisioning to try and connect via ssh without fiddling with keys at all since I've already set it up. To answer your question, in my use case you would know about it at the point that the parsing of the config.rb is done.

**Comment by jkeiser** [comment]
@squaresurf gotcha!  You might end up in good shape if you actually load them in config.rb ... but I can imagine us wanting to support ssh-agents too!  Thanks.

**Comment by squaresurf** [comment]
I couldn't load them in the config.rb since they're passphrase protected, which would require me to either hard code my password or ask for my passphrase every time I ran chef-provisioning. Both of those solutions aren't really acceptable in my opinion. I'm glad this was merged for others to use :+1:
